### PR TITLE
Biome rest api fixes

### PR DESCRIPTION
**Pull request by RyanLassigBanks**
Various fixes for biome rest api that were requested as part of stabilization.

### CODE REVIEW

File: libsplinter/src/biome/credentials/store/diesel/mod.rs [hunk 1]
```diff
@@ -39,7 +39,7 @@ impl SplinterCredentialsStore {
     /// # Arguments
     ///
     ///  * `connection_pool`: connection pool to the database
-    #[cfg(feature = "biome-rest-api")]
+    #[allow(dead_code)]
```
Comment (vaporos) [inline libsplinter/src/biome/credentials/store/diesel/mod.rs:42]: This was correct before -- the guard should just be removed when we remove biome-rest-api feature.

File: libsplinter/src/biome/rest_api/config.rs [hunk 2]
```diff
@@ -39,19 +39,27 @@ pub struct BiomeRestConfig {
 }
 
 impl BiomeRestConfig {
+    /// Returns token issuer string. Defaults to "self-issued"
```
Comment (vaporos) [inline libsplinter/src/biome/rest_api/config.rs:46]: I feel like if it has one period, then everything has to end in periods. Maybe: "Return token issuer string (default: "self-issued")."

File: libsplinter/src/biome/rest_api/mod.rs [hunk 3]
```diff
@@ -127,15 +127,10 @@ use crate::rest_api::sessions::AccessTokenIssuer;
 /// * `GET /biome/user/{id}` - Retrieve user with specified ID
 /// * `DELETE /biome/user/{id}` - Remove user with specified ID
 pub struct BiomeRestResourceManager {
-    // Disable lint warning, for now this is only used if the biome-credentials feature is enabled
```
Comment (vaporos) [inline libsplinter/src/biome/rest_api/mod.rs:109]: These should be changed to #[cfg(feature = "biome-rest-api")] until that feature is removed.

File: libsplinter/src/biome/rest_api/config.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (RyanLassigBanks) [reply libsplinter/src/biome/rest_api/config.rs:46]: The other doc comments in Biome are written in an [active voice](https://github.com/Cargill/splinter/blob/935d412c3f3c8370d18dfaefaca1eba597e542fd/libsplinter/src/biome/key_management/store/mod.rs#L34), so I'm trying to be consistent with those.

**Pull request merged**